frontend/src/pages/RunList.test.tsx

```diff
@@ -398,6 +398,126 @@ describe('RunList', () => {
+            Array [
+              Object {
+                "customRenderer": [Function],
+                "flex": 1.5,
+                "label": "Run name",
+                "sortKey": "name",
+              },
+              Object {
+                "customRenderer": [Function],
+                "flex": 0.5,
+                "label": "Status",
+              },
+              Object {
+                "flex": 0.5,
+                "label": "Duration",
+              },
+              Object {
+                "customRenderer": [Function],
+                "flex": 1,
+                "label": "Experiment",
+              },
+              Object {
+                "customRenderer": [Function],
+                "flex": 1,
+                "label": "Pipeline Version",
+              },
+              Object {
+                "customRenderer": [Function],
+                "flex": 0.5,
+                "label": "Recurring Run",
+              },
+              Object {
+                "flex": 1,
+                "label": "Start time",
+                "sortKey": "created_at",
+              },
+              Object {
+                "customRenderer": [Function],
+                "flex": 0.1,
+                "label": "",
+              },
+              Object {
+                "customRenderer": [Function],
+                "flex": 0.5,
+                "label": "z_metric",
+              },
+              Object {
+                "customRenderer": [Function],
+                "flex": 0.5,
+                "label": "y_metric",
+              },
+            ]
+          }
+          emptyMessage="No available runs found."
+          filterLabel="Filter runs"
+          initialSortColumn="created_at"
+          reload={[Function]}
+          rows={
+            Array [
+              Object {
+                "error": undefined,
+                "id": "testrun1",
+                "otherFields": Array [
+                  "run with id: testrun1",
+                  "-",
+                  "-",
+                  undefined,
+                  undefined,
+                  undefined,
+                  "-",
+                  "",
+                  Object {
+                    "metadata": Object {
+                      "count": 1,
+                      "maxValue": 2,
+                      "minValue": 2,
+                      "name": "z_metric",
+                    },
+                    "metric": Object {
+                      "name": "z_metric",
+                      "number_value": 2,
+                    },
+                  },
+                  Object {
+                    "metadata": Object {
+                      "count": 1,
+                      "maxValue": 1,
+                      "minValue": 1,
+                      "name": "y_metric",
+                    },
+                    "metric": Object {
+                      "name": "y_metric",
+                      "number_value": 1,
+                    },
```

Comment:
nit: the test is kind of verbose.
There are two typical ways to improve:
1. mock CustomTable to a simple component that renders the rows with no extra boilerplate, so snapshot is smaller
2. change shallow rendering to `@testing-library/react` and write selectors that verify order of the metrics.

I suggest 2, but I understand it can be too complex to first time contributors. Therefore, I think we can get this in first.

/lgtm
/cc @zijianjoy 
what do you think?

Reply:
I agree that it's verbose, I was trying to figure out a better way to just test the order of the metrics but wasn't able to find a good way to do it. 

I can try out your suggestion as part of this PR or as a separate PR. If part of this PR, just note that it might take me a bit since I'll need to ramp up. If you have any guidance on where to start or an example that I can reference, let me know! thanks

Reply:
I think there is an easier way, add a data-test-id to a metric value and select all Doms with the specified data-test-id using @testing-library/react.
Now we can convert each item to text and verify their order

Reply:
https://github.com/kubeflow/pipelines/blob/0fd1e6a52fe767d9798dcc927c07a7885ab9bb32/frontend/src/components/ResourceInfo.test.tsx#L79

You can refer to this as an example, also read react testing library documentation for this method -- getallbytestid

Reply:
thanks! will take a look and make changes

Reply:
Hey @Bobgy, since I am checking the order of `columns` in `CustomTable`, I am not sure there is a good way to add the `data-testid` to just the columns associated with the metrics. I can add the test id to the  `<CustomTable>` or for all columns in customtable but that doesn't help only select the metrics-related columns to verify the order.
https://github.com/kubeflow/pipelines/blob/58f74d33476f66e4de6fecee4bb1ffb82d3ef33a/frontend/src/pages/RunList.tsx#L177
https://github.com/kubeflow/pipelines/blob/58f74d33476f66e4de6fecee4bb1ffb82d3ef33a/frontend/src/components/CustomTable.tsx#L335

Did I misunderstand what you meant? I'm a little lost at this point, so any pointer would be appreciated. thanks!

Reply:
Hi @annajung, that was a good point. let me show you how to do this

When CustomTable renders columns, each column may use a custom renderer function, see the function for metric column: https://github.com/kubeflow/pipelines/blob/58f74d33476f66e4de6fecee4bb1ffb82d3ef33a/frontend/src/pages/RunList.tsx#L317-L326

Therefore, you can add data-testid in one of the divs inside the `Metric` component, see https://github.com/kubeflow/pipelines/blob/58f74d33476f66e4de6fecee4bb1ffb82d3ef33a/frontend/src/components/Metric.tsx#L93-L97

Reply:
Thanks @Bobgy, I did try that before but in the test, it would always fail to find it. It was due to my lack of knowledge in using mount vs shallow. I got it working with `mountWithRouter` and `data-testid` added to the Metric div. PTAL again when you get a chance!